torch/lib/THCUNN/generic/SpatialDepthwiseConvolution.cu

```diff
@@ -0,0 +1,201 @@
+#ifndef THC_GENERIC_FILE
+#define THC_GENERIC_FILE "generic/SpatialDepthwiseConvolution.cu"
+#else
+
+void THNN_(SpatialDepthwiseConvolution_updateOutput)(
+                  THCState *state,
+                  THCTensor *input,
+                  THCTensor *output,
+                  THCTensor *weight,
+                  THCTensor *bias,
+                  int kW, int kH,
+                  int dW, int dH,
+                  int padW, int padH,
+                  int dilationW, int dilationH)
+{
+  THCUNN_assertSameGPU(state, 3, input, output, weight);
+
+  // Only handle 4D Input Tensors for now
+  assert(THCTensor_(nDimension)(state, input) == 4);
+  assert(THCTensor_(nDimension)(state, weight) == 4);
+
+  // We assume that the input and weight Tensors are shaped properly by
+  // the caller, so we verify that here to some extent
+
+  // Weight Tensor is shape (output_channels, 1, kH, kW)
+  assert(weight->size[1] == 1);
+
+  // Input Tensor is shape (N, input_channels, H, W)
+  // We verify that the # of output_channels is a multiple of input_channels
+  assert(weight->size[0] % input->size[1] == 0);
+
+  // Bias has same # of channels as output
+  if (bias) {
+    assert(bias->size[0] == weight->size[0]);
```

Comment:
use `THAssert`. `assert` is a no-op if the library is compiled with `-DNDEBUG` (common for release builds)